### DISCUSSION TITLE
Quantity logging and other improvement suggestions

### DISCUSSION THREAD

**Discussion by gramapp** [jlai/fitness-dashboard#11] [General]
When entering food, quantity has to be initially selected from a drop down and only after that user is allowed to adjust the actual quantity. I am using chrome.

Is it possible to do it in the same way fit bit used to do? Enter quantity from keyboard and then select units from drop down. Its preferable to  be able to do that just with key board. (using arrows and tab buttons) to avoid wasting time on the mouse.

Is it possible to have 'when' field to default to the last selected setting in the current session.

Separate the check box and food name. When food name is clicked it should take user to the detailed nutritional information. https://www.fitbit.com/foods/food?viewFood=on&foodId=730349677 

![image](https://github.com/jlai/fitness-dashboard/assets/175133758/1f9ed9c6-5d36-4b18-8266-347cc8847c61)

All of the above behaviors were in fit bit online dashboard.

**Comment by gramapp** [comment]
Food search box on initial entry requires an enter key to search, but subsequent once don't. My preference is to use enter key for search. However if memory serves right fitbit dashboard would search on every key stroked... stupid.

**Comment by jlai** [comment]
The quantity selector was a bit buggy. I've made some improvements. Try again? You should be able to type in a number like "3" and then use the up/down arrow key to select units. Also changed it to remember the "when" setting.

According to the [documentation](https://dev.fitbit.com/build/reference/web-api/nutrition/get-food/#Additional-Information), nutrition information is only available for custom foods, not the public database, so unfortunately I don't think it'll be possible to show nutrition data for the majority of foods.

Food search is super buggy right now so it'll need some attention.

**Reply by gramapp** [reply]
Yes its working well. Thank you.

FYI food logging part of fitbit dashboard is working now. You may want to take a look an save their code for reference, before they take it down again.

https://www.fitbit.com/foods/log

**Reply by chrissard2** [reply]
after log you need to add a date like  

/2024/07/11

**Reply by gramapp** [reply]
> after log you need to add a date like
> 
> /2024/07/11

the link as i posted takes user to current date. its possible to append ymd to go to a particular date directly.

**Reply by gramapp** [reply]
https://www.fitbit.com/foods/log no longer works as of today ;-(

**Comment by gramapp** [comment]
> When entering food, quantity has to be initially selected from a drop down and only after that user is allowed to adjust the actual quantity. I am using chrome.
> 
> Is it possible to do it in the same way fit bit used to do? Enter quantity from keyboard and then select units from drop down. Its preferable to be able to do that just with key board. (using arrows and tab buttons) to avoid wasting time on the mouse.
> 
> Is it possible to have 'when' field to default to the last selected setting in the current session.
> 
> Separate the check box and food name. When food name is clicked it should take user to the detailed nutritional information. https://www.fitbit.com/foods/food?viewFood=on&foodId=730349677
> 
> ![image](https://private-user-images.githubusercontent.com/175133758/347486082-1f9ed9c6-5d36-4b18-8266-347cc8847c61.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwNTA4MjMsIm5iZiI6MTcyMTA1MDUyMywicGF0aCI6Ii8xNzUxMzM3NTgvMzQ3NDg2MDgyLTFmOWVkOWM2LTVkMzYtNGIxOC04MjY2LTM0N2NjODg0N2M2MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxNVQxMzM1MjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03YjdmYWE0MGIxODZhYjYwNDJiYmMxNzc2YmEzOTM4NzM1MTk3M2FiNWM3Mjc1MmRiYmJjMzA1ZDljN2NmZTM1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.YYZcpO6VYQtKamhhakda1wAJcvrJomTWBhxshzng4TA)
> 
> All of the above behaviors were in fit bit online dashboard.

Jlai, there is some sort of an issue entering values of less than 1. For example, I am not able to enter .5 cookie or .5 bar. instead I have to enter half value equivalent in grams.

**Reply by jlai** [reply]
0.5 works, but I've updated it to accept .5
